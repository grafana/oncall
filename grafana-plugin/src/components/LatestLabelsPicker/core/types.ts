export interface ItemRepresentation {
  id: string;
  repr: string;
}

export interface ItemGroup {
  key: ItemRepresentation;
  values: ItemRepresentation[];
}

export interface ItemSelected {
  key: ItemRepresentation;
  value: ItemRepresentation;
}