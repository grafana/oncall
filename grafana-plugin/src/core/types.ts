export interface ItemGroup {
  key: any;
  values: any[];
}

export interface ItemSelected {
  key: any;
  value: any;
}

export enum LabelInputType {
  key = 'key',
  value = 'value',
}

export interface ServiceLabelValidator {
  isValid: boolean;
  errorMessage?: string;
}
