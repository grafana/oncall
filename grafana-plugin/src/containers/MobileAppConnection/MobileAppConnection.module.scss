.container {
  display: flex;
  flex-direction: row;
  min-width: 100%;

  &__box {
    flex-basis: 50%;
  }

  &__box:first-child {
    margin-right: 8px;
  }

  &__box:last-child {
    margin-left: 8px;
  }
}

.notification-buttons {
  width: 100%;
  padding-top: 12px;
}

.icon {
  margin-top: -6px;
  margin-left: 4px;
  fill: var(--green-6);
}

.disconnect__container {
  position: relative;
  display: flex;
  justify-content: center;
  width: 100%;
}

.disconnect__qrCode {
  width: 240px;
  height: auto;
  filter: blur(6px);
  opacity: 0.6;
}

.blurry {
  filter: blur(4px);
  opacity: 0.2;
}

.qr-code {
  background-color: #fff;
  margin-bottom: 12px;
}

.qr-loader {
  position: absolute;
  z-index: 10;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  text-align: center;

  &__text {
    text-align: center;
    margin-bottom: 12px;
    display: block;
  }

  i {
    // Overwrite Grafana's loading icon
    font-size: 32px;
  }
}
