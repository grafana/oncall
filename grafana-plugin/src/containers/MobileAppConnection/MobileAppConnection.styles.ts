import { css } from '@emotion/css';
import { Colors } from 'styles/utils.styles';

export const getMobileAppConnectionStyles = () => {
  return {
    container: css`
      display: flex;
      flex-direction: row;
      min-width: 100%;

      @media (max-width: 768px) {
        flex-direction: column;
      }
    `,

    containerBox: css`
      flex-basis: 50%;

      &:first-child {
        margin-right: 8px;
      }
      *:last-child {
        margin-left: 8px;
      }

      @media (max-width: 768px) {
        &:first-child {
          margin-right: 0px;
          margin-bottom: 8px;
        }
        &:last-child {
          margin-left: 0px;
          margin-top: 8px;
        }
      }
    `,

    notificationButtons: css`
      width: 100%;
      padding-top: 12px;
    `,

    icon: css`
      margin-top: -6px;
      margin-left: 4px;
      fill: ${Colors.GREEN_6};
    `,

    disconnectContainer: css`
      position: relative;
      display: flex;
      justify-content: center;
      width: 100%;
    `,
    disconnectQRCode: css`
      width: 240px;
      height: auto;
      filter: blur(6px);
      opacity: 0.6;
    `,

    blurry: css`
      filter: blur(4px);
      opacity: 0.2;
    `,

    qrCode: css`
      background-color: #fff;
      margin-bottom: 12px;
    `,

    qrLoader: css`
      position: absolute;
      z-index: 10;
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%);
      text-align: center;

      i {
        font-size: 32px;
      }
    `,

    qrLoaderText: css`
      text-align: center;
      margin-bottom: 12px;
      display: block;
    `,
  };
};
