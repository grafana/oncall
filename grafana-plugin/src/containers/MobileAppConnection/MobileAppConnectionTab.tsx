import React from 'react';

import { observer } from 'mobx-react';

import { Text } from 'components/Text/Text';

const MobileAppConnectionTab: React.FC<{ userPk?: string }> = observer(() => {
  return (
    <Text type="secondary">
      Mobile settings have been moved to{' '}
      <a href={`${window.location.origin}/profile?tab=irm`}>
        <Text type="link">user's profile</Text>
      </a>
    </Text>
  );
});

export { MobileAppConnectionTab };
