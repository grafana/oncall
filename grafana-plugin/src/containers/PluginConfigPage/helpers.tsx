export const constructSyncErrorMessage = (errMsg: string, url?: string): string => `${url ? `${url}\n` : ''}${errMsg}`;

export const constructErrorActionMessage = (msg?: string): string =>
  `Try removing your current configuration, ${
    msg ? msg : 'double checking your settings'
  }, and re-initializing the plugin.\nBy removing your current configuration, you will need to ensure that you regenerate a new invite token, and input this in your new configuration.`;
