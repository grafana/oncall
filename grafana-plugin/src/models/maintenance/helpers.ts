import moment from 'moment-timezone';

import { Maintenance } from './maintenance.types';

export function getMaintenanceProgress(maintenance: Maintenance) {
  const duration = maintenance.maintenance_till_timestamp - maintenance.started_at_timestamp;
  const passed = moment().unix() - maintenance.started_at_timestamp;

  return (passed / duration) * 100;
}
