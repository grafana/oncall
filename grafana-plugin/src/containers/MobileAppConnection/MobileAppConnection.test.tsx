import React from 'react';

import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { MemoryRouter } from 'react-router-dom';

import { UserHelper } from 'models/user/user.helpers';
import { ApiSchemas } from 'network/oncall-api/api.types';
import { rootStore } from 'state/rootStore';

import { MobileAppConnection } from './MobileAppConnection';

jest.mock('plugin/GrafanaPluginRootPage.helpers', () => ({
  isTopNavbar: () => false,
}));

jest.mock('utils/authorization/authorization', () => ({
  ...jest.requireActual('utils/authorization/authorization'),
  isUserActionAllowed: jest.fn().mockReturnValue(true),
}));

const loadUserMock = jest.fn().mockReturnValue(undefined);

jest.mock('state/rootStore', () => ({
  rootStore: jest.fn(),
}));

const mockRootStore = (rest?: any, connected = false, cloud_connected = true) => {
  rootStore.userStore = {
    fetchItemById: loadUserMock,
    currentUser: {
      messaging_backends: {
        MOBILE_APP: { connected },
      },
    } as unknown as ApiSchemas['User'],
    ...(rest ? rest : {}),
  };

  rootStore.cloudStore = {
    getCloudConnectionStatus: jest.fn().mockReturnValue({ cloud_connection_status: cloud_connected }),
    cloudConnectionStatus: { cloud_connection_status: cloud_connected },
  } as any;

  // @ts-ignore
  rootStore.isOpenSource = jest.fn().mockReturnValue(true);
  rootStore.hasFeature = jest.fn().mockReturnValue(true);
};

const USER_PK = '8585';
const BACKEND = 'MOBILE_APP';

describe('MobileAppConnection', () => {
  test('', () => {
    expect(true).toBe(true);
  });
});

describe('MobileAppConnection', () => {
  beforeEach(() => {
    loadUserMock.mockClear();
    (rootStore as any).mockClear();
    mockRootStore();
    UserHelper.fetchBackendConfirmationCode = jest.fn().mockResolvedValueOnce('dfd');
  });

  test('it shows a loading message if it is currently fetching the QR code', async () => {
    const component = render(<MobileAppConnection userPk={USER_PK} />);
    expect(component.container).toMatchSnapshot();

    await waitFor(() => {
      expect(UserHelper.fetchBackendConfirmationCode).toHaveBeenCalledTimes(1);
      expect(UserHelper.fetchBackendConfirmationCode).toHaveBeenCalledWith(USER_PK, BACKEND);
    });
  });

  test('it shows an error message if there was an error fetching the QR code', async () => {
    UserHelper.fetchBackendConfirmationCode = jest.fn().mockRejectedValueOnce('dfd');
    const component = render(<MobileAppConnection userPk={USER_PK} />);
    await screen.findByText(/.*error fetching your QR code.*/);

    await waitFor(() => {
      expect(component.container).toMatchSnapshot();

      expect(UserHelper.fetchBackendConfirmationCode).toHaveBeenCalledTimes(1);
      expect(UserHelper.fetchBackendConfirmationCode).toHaveBeenCalledWith(USER_PK, BACKEND);
    });
  });

  test("it shows a QR code if the app isn't already connected", async () => {
    const component = render(<MobileAppConnection userPk={USER_PK} />);
    expect(component.container).toMatchSnapshot();

    await waitFor(() => {
      expect(UserHelper.fetchBackendConfirmationCode).toHaveBeenCalledTimes(1);
      expect(UserHelper.fetchBackendConfirmationCode).toHaveBeenCalledWith(USER_PK, BACKEND);
    });
  });

  test('if we disconnect the app, it disconnects and fetches a new QR code', async () => {
    mockRootStore(
      {
        unlinkBackend: jest.fn().mockResolvedValueOnce('asdfadsfafds'),
      },
      true
    );

    const component = render(<MobileAppConnection userPk={USER_PK} />);
    const button = await screen.findByRole('button');

    // click the disconnect button, which opens the modal
    await userEvent.click(button);
    // click the confirm button within the modal, which actually triggers the callback
    await userEvent.click(screen.getByText('Remove'));

    expect(component.container).toMatchSnapshot();

    await waitFor(() => {
      expect(UserHelper.fetchBackendConfirmationCode).toHaveBeenCalledTimes(1);
      expect(UserHelper.fetchBackendConfirmationCode).toHaveBeenCalledWith(USER_PK, BACKEND);

      expect(rootStore.userStore.unlinkBackend).toHaveBeenCalledTimes(1);
      expect(rootStore.userStore.unlinkBackend).toHaveBeenCalledWith(USER_PK, BACKEND);
    });
  });

  test('it shows a loading message if it is currently disconnecting', async () => {
    mockRootStore(
      {
        unlinkBackend: jest.fn().mockResolvedValueOnce(new Promise((resolve) => setTimeout(resolve, 500))),
      },
      true
    );

    const component = render(<MobileAppConnection userPk={USER_PK} />);
    const button = await screen.findByRole('button');

    // click the disconnect button, which opens the modal
    await userEvent.click(button);
    // click the confirm button within the modal, which actually triggers the callback
    await userEvent.click(screen.getByText('Remove'));

    // wait for loading state
    await screen.findByText(/.*Loading.*/);

    expect(component.container).toMatchSnapshot();

    await waitFor(() => {
      expect(UserHelper.fetchBackendConfirmationCode).toHaveBeenCalledTimes(1);
      expect(UserHelper.fetchBackendConfirmationCode).toHaveBeenCalledWith(USER_PK, BACKEND);

      expect(rootStore.userStore.unlinkBackend).toHaveBeenCalledTimes(1);
      expect(rootStore.userStore.unlinkBackend).toHaveBeenCalledWith(USER_PK, BACKEND);
    });
  });

  test('it shows an error message if there was an error disconnecting the mobile app', async () => {
    mockRootStore(
      {
        unlinkBackend: jest.fn().mockRejectedValueOnce('asdfadsfafds'),
      },
      true
    );

    const component = render(<MobileAppConnection userPk={USER_PK} />);
    const button = await screen.findByTestId('test__disconnect');

    // click the disconnect button, which opens the modal
    await userEvent.click(button);
    // click the confirm button within the modal, which actually triggers the callback
    await userEvent.click(screen.getByText('Remove'));

    await screen.findByText(/.*error disconnecting your mobile app.*/);

    expect(component.container).toMatchSnapshot();

    await waitFor(() => {
      expect(UserHelper.fetchBackendConfirmationCode).toHaveBeenCalledTimes(0);

      expect(rootStore.userStore.unlinkBackend).toHaveBeenCalledTimes(1);
      expect(rootStore.userStore.unlinkBackend).toHaveBeenCalledWith(USER_PK, BACKEND);
    });
  });

  test('it polls loadUser on first render if not connected', async () => {
    mockRootStore(
      {
        unlinkBackend: jest.fn().mockRejectedValueOnce('asdfadsfafds'),
      },
      false
    );

    render(<MobileAppConnection userPk={USER_PK} />);

    await waitFor(
      () => {
        expect(loadUserMock).toHaveBeenCalled();
      },
      { timeout: 6000 }
    );
  });

  test('it polls loadUser after disconnect', async () => {
    mockRootStore(
      {
        unlinkBackend: jest.fn().mockRejectedValueOnce('asdff'),
      },
      true
    );

    render(<MobileAppConnection userPk={USER_PK} />);
    const button = await screen.findByRole('button');

    loadUserMock.mockClear();

    await userEvent.click(button); // click the disconnect button, which opens the modal
    await userEvent.click(screen.getByText('Remove')); // click the confirm button within the modal, which actually triggers the callback

    await waitFor(
      () => {
        expect(loadUserMock).toHaveBeenCalled();
      },
      { timeout: 6000 }
    );
  });

  test('it shows a warning when cloud is not connected', async () => {
    mockRootStore({}, true, false);

    // Using MemoryRouter to avoid "Invariant failed: You should not use <Link> outside a <Router>"
    const component = render(
      <MemoryRouter>
        <MobileAppConnection userPk={USER_PK} />
      </MemoryRouter>
    );
    expect(component.container).toMatchSnapshot();
  });
});
