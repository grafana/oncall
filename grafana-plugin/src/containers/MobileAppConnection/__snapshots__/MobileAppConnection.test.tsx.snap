// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileAppConnection if we disconnect the app, it disconnects and fetches a new QR code 1`] = `
<div>
  <div
    class="css-gjl87o-vertical-group"
    style="width: 100%; height: 100%;"
  >
    <div
      class="css-gxt817-layoutChildrenWrapper"
    >
      <div
        class="container"
      >
        <div
          class="root root_bordered root_shadowed root--withBackGround container__box"
        >
          <div
            class="css-gjl87o-vertical-group"
            style="width: 100%; height: 100%;"
          >
            <div
              class="css-12oo3x0-layoutChildrenWrapper"
            >
              <span
                class="root text text--primary text--medium text--strong"
              >
                Download
              </span>
            </div>
            <div
              class="css-12oo3x0-layoutChildrenWrapper"
            >
              <span
                class="root text text--primary text--medium"
              >
                The Grafana OnCall app is available on both the App Store and Google Play Store.
              </span>
            </div>
            <div
              class="css-12oo3x0-layoutChildrenWrapper"
            >
              <div
                class="css-gjl87o-vertical-group"
                style="width: 100%; height: 100%;"
              >
                <div
                  class="css-gxt817-layoutChildrenWrapper"
                >
                  <a
                    href="https://apps.apple.com/us/app/grafana-oncall-preview/id1669759048"
                    rel="noreferrer"
                    style="width: 100%;"
                    target="_blank"
                  >
                    <div
                      class="root root_bordered root--fullWidth root--withBackGround root--hover icon-block"
                    >
                      <img
                        alt="Apple"
                        class="icon"
                        src="[object Object]"
                      />
                      <span
                        class="root text text--primary text--medium icon-text"
                      >
                        iOS
                      </span>
                    </div>
                  </a>
                </div>
                <div
                  class="css-gxt817-layoutChildrenWrapper"
                >
                  <a
                    href="https://play.google.com/store/apps/details?id=com.grafana.oncall.prod"
                    rel="noreferrer"
                    style="width: 100%;"
                    target="_blank"
                  >
                    <div
                      class="root root_bordered root--fullWidth root--hover icon-block"
                    >
                      <img
                        alt="Play Store"
                        class="icon"
                        src="[object Object]"
                      />
                      <span
                        class="root text text--primary text--medium icon-text"
                      >
                        Android
                      </span>
                    </div>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="root root_bordered root_shadowed root--withBackGround container__box"
        >
          <div
            class="css-gjl87o-vertical-group"
            style="width: 100%; height: 100%;"
          >
            <div
              class="css-12oo3x0-layoutChildrenWrapper"
            >
              <span
                class="root text text--primary text--medium text--strong"
              >
                Sign In
              </span>
            </div>
            <div
              class="css-12oo3x0-layoutChildrenWrapper"
            >
              <span
                class="root text text--primary text--medium"
              >
                Open the Grafana OnCall mobile application and scan this code to sync it with your account.
              </span>
            </div>
            <div
              class="css-12oo3x0-layoutChildrenWrapper"
            >
              <div
                class="u-width-100 u-flex u-flex-center u-position-relative"
              >
                <div
                  class="root root_bordered qr-code"
                >
                  <svg
                    height="256"
                    viewBox="0 0 21 21"
                    width="256"
                  >
                    <path
                      d="M0,0 h21v21H0z"
                      fill="#FFFFFF"
                      shape-rendering="crispEdges"
                    />
                    <path
                      d="M0 0h7v1H0zM9 0h3v1H9zM14,0 h7v1H14zM0 1h1v1H0zM6 1h1v1H6zM8 1h2v1H8zM12 1h1v1H12zM14 1h1v1H14zM20,1 h1v1H20zM0 2h1v1H0zM2 2h3v1H2zM6 2h1v1H6zM8 2h1v1H8zM11 2h2v1H11zM14 2h1v1H14zM16 2h3v1H16zM20,2 h1v1H20zM0 3h1v1H0zM2 3h3v1H2zM6 3h1v1H6zM8 3h1v1H8zM10 3h2v1H10zM14 3h1v1H14zM16 3h3v1H16zM20,3 h1v1H20zM0 4h1v1H0zM2 4h3v1H2zM6 4h1v1H6zM8 4h1v1H8zM10 4h2v1H10zM14 4h1v1H14zM16 4h3v1H16zM20,4 h1v1H20zM0 5h1v1H0zM6 5h1v1H6zM8 5h1v1H8zM10 5h1v1H10zM12 5h1v1H12zM14 5h1v1H14zM20,5 h1v1H20zM0 6h7v1H0zM8 6h1v1H8zM10 6h1v1H10zM12 6h1v1H12zM14,6 h7v1H14zM10 7h2v1H10zM2 8h1v1H2zM5 8h5v1H5zM11 8h1v1H11zM13 8h1v1H13zM15 8h5v1H15zM5 9h1v1H5zM9 9h1v1H9zM13 9h1v1H13zM15 9h1v1H15zM17 9h1v1H17zM20,9 h1v1H20zM0 10h3v1H0zM4 10h1v1H4zM6 10h5v1H6zM13 10h5v1H13zM20,10 h1v1H20zM0 11h1v1H0zM2 11h2v1H2zM7 11h1v1H7zM10 11h1v1H10zM12 11h4v1H12zM17 11h1v1H17zM2 12h2v1H2zM5 12h7v1H5zM13 12h5v1H13zM20,12 h1v1H20zM8 13h1v1H8zM10 13h1v1H10zM17 13h1v1H17zM20,13 h1v1H20zM0 14h7v1H0zM8 14h7v1H8zM17 14h2v1H17zM20,14 h1v1H20zM0 15h1v1H0zM6 15h1v1H6zM8 15h2v1H8zM12 15h3v1H12zM16 15h1v1H16zM19,15 h2v1H19zM0 16h1v1H0zM2 16h3v1H2zM6 16h1v1H6zM9 16h2v1H9zM13 16h1v1H13zM17,16 h4v1H17zM0 17h1v1H0zM2 17h3v1H2zM6 17h1v1H6zM9 17h1v1H9zM12 17h2v1H12zM15 17h1v1H15zM17 17h2v1H17zM0 18h1v1H0zM2 18h3v1H2zM6 18h1v1H6zM8 18h1v1H8zM10 18h2v1H10zM13 18h1v1H13zM15 18h3v1H15zM19,18 h2v1H19zM0 19h1v1H0zM6 19h1v1H6zM10 19h1v1H10zM13 19h3v1H13zM17 19h2v1H17zM0 20h7v1H0zM10 20h1v1H10zM14 20h5v1H14zM20,20 h1v1H20z"
                      fill="#000000"
                      shape-rendering="crispEdges"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MobileAppConnection it shows a QR code if the app isn't already connected 1`] = `
<div>
  <div
    class="css-gjl87o-vertical-group"
    style="width: 100%; height: 100%;"
  >
    <div
      class="css-gxt817-layoutChildrenWrapper"
    >
      <div
        class="container"
      >
        <div
          class="root root_bordered root_shadowed root--withBackGround container__box"
        >
          <div
            class="css-gjl87o-vertical-group"
            style="width: 100%; height: 100%;"
          >
            <div
              class="css-12oo3x0-layoutChildrenWrapper"
            >
              <span
                class="root text text--primary text--medium text--strong"
              >
                Download
              </span>
            </div>
            <div
              class="css-12oo3x0-layoutChildrenWrapper"
            >
              <span
                class="root text text--primary text--medium"
              >
                The Grafana OnCall app is available on both the App Store and Google Play Store.
              </span>
            </div>
            <div
              class="css-12oo3x0-layoutChildrenWrapper"
            >
              <div
                class="css-gjl87o-vertical-group"
                style="width: 100%; height: 100%;"
              >
                <div
                  class="css-gxt817-layoutChildrenWrapper"
                >
                  <a
                    href="https://apps.apple.com/us/app/grafana-oncall-preview/id1669759048"
                    rel="noreferrer"
                    style="width: 100%;"
                    target="_blank"
                  >
                    <div
                      class="root root_bordered root--fullWidth root--withBackGround root--hover icon-block"
                    >
                      <img
                        alt="Apple"
                        class="icon"
                        src="[object Object]"
                      />
                      <span
                        class="root text text--primary text--medium icon-text"
                      >
                        iOS
                      </span>
                    </div>
                  </a>
                </div>
                <div
                  class="css-gxt817-layoutChildrenWrapper"
                >
                  <a
                    href="https://play.google.com/store/apps/details?id=com.grafana.oncall.prod"
                    rel="noreferrer"
                    style="width: 100%;"
                    target="_blank"
                  >
                    <div
                      class="root root_bordered root--fullWidth root--hover icon-block"
                    >
                      <img
                        alt="Play Store"
                        class="icon"
                        src="[object Object]"
                      />
                      <span
                        class="root text text--primary text--medium icon-text"
                      >
                        Android
                      </span>
                    </div>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="root root_bordered root_shadowed root--withBackGround container__box"
        >
          <div
            class="css-1yjvs5a"
          >
            Loading...
             
            <div
              class="css-1tqtz24"
              data-testid="Spinner"
            >
              <i
                aria-label="loading spinner"
                class="fa fa-spinner fa-spin fa-spin"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MobileAppConnection it shows a loading message if it is currently disconnecting 1`] = `
<div>
  <div
    class="css-gjl87o-vertical-group"
    style="width: 100%; height: 100%;"
  >
    <div
      class="css-gxt817-layoutChildrenWrapper"
    >
      <div
        class="container"
      >
        <div
          class="root root_bordered root_shadowed root--withBackGround container__box"
        >
          <div
            class="css-gjl87o-vertical-group"
            style="width: 100%; height: 100%;"
          >
            <div
              class="css-12oo3x0-layoutChildrenWrapper"
            >
              <span
                class="root text text--primary text--medium text--strong"
              >
                Download
              </span>
            </div>
            <div
              class="css-12oo3x0-layoutChildrenWrapper"
            >
              <span
                class="root text text--primary text--medium"
              >
                The Grafana OnCall app is available on both the App Store and Google Play Store.
              </span>
            </div>
            <div
              class="css-12oo3x0-layoutChildrenWrapper"
            >
              <div
                class="css-gjl87o-vertical-group"
                style="width: 100%; height: 100%;"
              >
                <div
                  class="css-gxt817-layoutChildrenWrapper"
                >
                  <a
                    href="https://apps.apple.com/us/app/grafana-oncall-preview/id1669759048"
                    rel="noreferrer"
                    style="width: 100%;"
                    target="_blank"
                  >
                    <div
                      class="root root_bordered root--fullWidth root--withBackGround root--hover icon-block"
                    >
                      <img
                        alt="Apple"
                        class="icon"
                        src="[object Object]"
                      />
                      <span
                        class="root text text--primary text--medium icon-text"
                      >
                        iOS
                      </span>
                    </div>
                  </a>
                </div>
                <div
                  class="css-gxt817-layoutChildrenWrapper"
                >
                  <a
                    href="https://play.google.com/store/apps/details?id=com.grafana.oncall.prod"
                    rel="noreferrer"
                    style="width: 100%;"
                    target="_blank"
                  >
                    <div
                      class="root root_bordered root--fullWidth root--hover icon-block"
                    >
                      <img
                        alt="Play Store"
                        class="icon"
                        src="[object Object]"
                      />
                      <span
                        class="root text text--primary text--medium icon-text"
                      >
                        Android
                      </span>
                    </div>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="root root_bordered root_shadowed root--withBackGround container__box"
        >
          <div
            class="css-1yjvs5a"
          >
            Loading...
             
            <div
              class="css-1tqtz24"
              data-testid="Spinner"
            >
              <i
                aria-label="loading spinner"
                class="fa fa-spinner fa-spin fa-spin"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MobileAppConnection it shows a loading message if it is currently fetching the QR code 1`] = `
<div>
  <div
    class="css-gjl87o-vertical-group"
    style="width: 100%; height: 100%;"
  >
    <div
      class="css-gxt817-layoutChildrenWrapper"
    >
      <div
        class="container"
      >
        <div
          class="root root_bordered root_shadowed root--withBackGround container__box"
        >
          <div
            class="css-gjl87o-vertical-group"
            style="width: 100%; height: 100%;"
          >
            <div
              class="css-12oo3x0-layoutChildrenWrapper"
            >
              <span
                class="root text text--primary text--medium text--strong"
              >
                Download
              </span>
            </div>
            <div
              class="css-12oo3x0-layoutChildrenWrapper"
            >
              <span
                class="root text text--primary text--medium"
              >
                The Grafana OnCall app is available on both the App Store and Google Play Store.
              </span>
            </div>
            <div
              class="css-12oo3x0-layoutChildrenWrapper"
            >
              <div
                class="css-gjl87o-vertical-group"
                style="width: 100%; height: 100%;"
              >
                <div
                  class="css-gxt817-layoutChildrenWrapper"
                >
                  <a
                    href="https://apps.apple.com/us/app/grafana-oncall-preview/id1669759048"
                    rel="noreferrer"
                    style="width: 100%;"
                    target="_blank"
                  >
                    <div
                      class="root root_bordered root--fullWidth root--withBackGround root--hover icon-block"
                    >
                      <img
                        alt="Apple"
                        class="icon"
                        src="[object Object]"
                      />
                      <span
                        class="root text text--primary text--medium icon-text"
                      >
                        iOS
                      </span>
                    </div>
                  </a>
                </div>
                <div
                  class="css-gxt817-layoutChildrenWrapper"
                >
                  <a
                    href="https://play.google.com/store/apps/details?id=com.grafana.oncall.prod"
                    rel="noreferrer"
                    style="width: 100%;"
                    target="_blank"
                  >
                    <div
                      class="root root_bordered root--fullWidth root--hover icon-block"
                    >
                      <img
                        alt="Play Store"
                        class="icon"
                        src="[object Object]"
                      />
                      <span
                        class="root text text--primary text--medium icon-text"
                      >
                        Android
                      </span>
                    </div>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="root root_bordered root_shadowed root--withBackGround container__box"
        >
          <div
            class="css-1yjvs5a"
          >
            Loading...
             
            <div
              class="css-1tqtz24"
              data-testid="Spinner"
            >
              <i
                aria-label="loading spinner"
                class="fa fa-spinner fa-spin fa-spin"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MobileAppConnection it shows a warning when cloud is not connected 1`] = `
<div>
  <div
    class="css-gjl87o-vertical-group"
    style="width: 100%; height: 100%;"
  >
    <div
      class="css-12oo3x0-layoutChildrenWrapper"
    >
      <span
        class="root text text--secondary text--medium"
      >
        Please connect Grafana Cloud OnCall to use the mobile app
      </span>
    </div>
    <div
      class="css-12oo3x0-layoutChildrenWrapper"
    >
      <a
        class="root"
        href="/a/grafana-oncall-app/cloud"
      >
        <button
          class="css-8b29hm-button"
          type="button"
        >
          <div
            class="css-1j2891d-Icon"
          />
          <span
            class="css-1riaxdn"
          >
            Connect Grafana Cloud OnCall
          </span>
        </button>
      </a>
    </div>
  </div>
</div>
`;

exports[`MobileAppConnection it shows an error message if there was an error disconnecting the mobile app 1`] = `
<div>
  <div
    class="css-gjl87o-vertical-group"
    style="width: 100%; height: 100%;"
  >
    <div
      class="css-gxt817-layoutChildrenWrapper"
    >
      <div
        class="container"
      >
        <div
          class="root root_bordered root_shadowed root--withBackGround container__box"
        >
          <div
            class="css-gjl87o-vertical-group"
            style="width: 100%; height: 100%;"
          >
            <div
              class="css-12oo3x0-layoutChildrenWrapper"
            >
              <span
                class="root text text--primary text--medium text--strong"
              >
                Download
              </span>
            </div>
            <div
              class="css-12oo3x0-layoutChildrenWrapper"
            >
              <span
                class="root text text--primary text--medium"
              >
                The Grafana OnCall app is available on both the App Store and Google Play Store.
              </span>
            </div>
            <div
              class="css-12oo3x0-layoutChildrenWrapper"
            >
              <div
                class="css-gjl87o-vertical-group"
                style="width: 100%; height: 100%;"
              >
                <div
                  class="css-gxt817-layoutChildrenWrapper"
                >
                  <a
                    href="https://apps.apple.com/us/app/grafana-oncall-preview/id1669759048"
                    rel="noreferrer"
                    style="width: 100%;"
                    target="_blank"
                  >
                    <div
                      class="root root_bordered root--fullWidth root--withBackGround root--hover icon-block"
                    >
                      <img
                        alt="Apple"
                        class="icon"
                        src="[object Object]"
                      />
                      <span
                        class="root text text--primary text--medium icon-text"
                      >
                        iOS
                      </span>
                    </div>
                  </a>
                </div>
                <div
                  class="css-gxt817-layoutChildrenWrapper"
                >
                  <a
                    href="https://play.google.com/store/apps/details?id=com.grafana.oncall.prod"
                    rel="noreferrer"
                    style="width: 100%;"
                    target="_blank"
                  >
                    <div
                      class="root root_bordered root--fullWidth root--hover icon-block"
                    >
                      <img
                        alt="Play Store"
                        class="icon"
                        src="[object Object]"
                      />
                      <span
                        class="root text text--primary text--medium icon-text"
                      >
                        Android
                      </span>
                    </div>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="root root_bordered root_shadowed root--withBackGround container__box"
        >
          <span
            class="root text text--primary text--medium"
          >
            There was an error disconnecting your mobile app. Please try again.
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MobileAppConnection it shows an error message if there was an error fetching the QR code 1`] = `
<div>
  <div
    class="css-gjl87o-vertical-group"
    style="width: 100%; height: 100%;"
  >
    <div
      class="css-gxt817-layoutChildrenWrapper"
    >
      <div
        class="container"
      >
        <div
          class="root root_bordered root_shadowed root--withBackGround container__box"
        >
          <div
            class="css-gjl87o-vertical-group"
            style="width: 100%; height: 100%;"
          >
            <div
              class="css-12oo3x0-layoutChildrenWrapper"
            >
              <span
                class="root text text--primary text--medium text--strong"
              >
                Download
              </span>
            </div>
            <div
              class="css-12oo3x0-layoutChildrenWrapper"
            >
              <span
                class="root text text--primary text--medium"
              >
                The Grafana OnCall app is available on both the App Store and Google Play Store.
              </span>
            </div>
            <div
              class="css-12oo3x0-layoutChildrenWrapper"
            >
              <div
                class="css-gjl87o-vertical-group"
                style="width: 100%; height: 100%;"
              >
                <div
                  class="css-gxt817-layoutChildrenWrapper"
                >
                  <a
                    href="https://apps.apple.com/us/app/grafana-oncall-preview/id1669759048"
                    rel="noreferrer"
                    style="width: 100%;"
                    target="_blank"
                  >
                    <div
                      class="root root_bordered root--fullWidth root--withBackGround root--hover icon-block"
                    >
                      <img
                        alt="Apple"
                        class="icon"
                        src="[object Object]"
                      />
                      <span
                        class="root text text--primary text--medium icon-text"
                      >
                        iOS
                      </span>
                    </div>
                  </a>
                </div>
                <div
                  class="css-gxt817-layoutChildrenWrapper"
                >
                  <a
                    href="https://play.google.com/store/apps/details?id=com.grafana.oncall.prod"
                    rel="noreferrer"
                    style="width: 100%;"
                    target="_blank"
                  >
                    <div
                      class="root root_bordered root--fullWidth root--hover icon-block"
                    >
                      <img
                        alt="Play Store"
                        class="icon"
                        src="[object Object]"
                      />
                      <span
                        class="root text text--primary text--medium icon-text"
                      >
                        Android
                      </span>
                    </div>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="root root_bordered root_shadowed root--withBackGround container__box"
        >
          <span
            class="root text text--primary text--medium"
          >
            There was an error fetching your QR code. Please try again.
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
